### DISCUSSION TITLE
How to get the image controls when the image is also a link?

### DISCUSSION THREAD

**Discussion by LorenAmelang** [TriliumNext/Trilium#3836] [Q&A]
I run into this occasionally...  I save a web page, and an included image is also a URL link. If I click the image in Trilium without being in edit mode, the web page opens. 

In edit mode, if I want to, say, reduce the size of the image, I seem to almost always get the link control popup: 

![Trilium link controls](https://user-images.githubusercontent.com/1297090/231621206-574f13c2-fe57-42ad-bccb-8283ac430886.jpg)

If I poke around long enough, I can usually get the image controls, but I'm not sure how it happens, and it is very fragile - most any action will revert to the link controls: 

![Trilium hard to delete image that is also a link](https://user-images.githubusercontent.com/1297090/231621399-cfbbf49e-87e2-4249-9702-1ebb01b48147.jpg)

Is there some clear tactic for accessing the image options?

**Comment by maphew** [comment]
There's the inverse problem also: how to get at Link properties for an image when the image properties are active?

**Comment by LorenAmelang** [comment]
Just tried this again, in 0.63.5. When I click in the image, the image properties bar flashes above the image, then disappears and the link properties bar opens under the image. If I then press the Esc key, the link properties bar is replaced by the image properties bar, now below the image. Seems useful...  

@maphew  No, it doesn't seem to go back to link properties if I Esc again. But maybe it would swap once if you get image properties first? 

But if I move the mouse while making that initial click in the image, it draws a small floating version of the image on top of the full version, and when I release the click the whole image disappears! Thankfully Ctrl+Z will bring it back.

**Reply by maphew** [reply]
Thanks for the tip about Esc, I can use that.

The only way I've discovered so far to have an image with a link is to create that in something else and then paste the image+link into Trilium. After that I can get to the link properties popup as described above.
